logpyle/runalyzer.py

```diff
@@ -55,16 +55,19 @@ def get_rank_agg_table(self, qty, rank_aggregator):
         self.rank_agg_tables.add((qty, rank_aggregator))
         return tbl_name
 
-    def scatter_cursor(self, cursor, *args, **kwargs):
-        from matplotlib.pyplot import scatter, show
+    def scatter_cursor(self, cursor, labels=[], *args, **kwargs):
+        import matplotlib.pyplot as plt
 
         data_args = tuple(zip(*list(cursor)))
-        scatter(*(data_args + args), **kwargs)
+        plt.scatter(*(data_args + args), **kwargs)
+
+        plt.xlabel(labels.pop(0) if labels else "")
+        plt.ylabel(labels.pop(0) if labels else "")
 
         if self.interactive:
-            show()
+            plt.show()
 
-    def plot_cursor(self, cursor, *args, **kwargs):
+    def plot_cursor(self, cursor, labels=[], *args, **kwargs):
```

Comment:
Should check whether non-`None` labels has correct form and assign X/Y label variables during input normalization.

Reply:
Not 100% what you had in mind here but I implemented something in this direction.